src/vm/analysis/read_only_checker/tests.rs

```diff
@@ -3,6 +3,22 @@ use vm::database::{MemoryBackingStore};
 use vm::analysis::{type_check, mem_type_check, CheckError, CheckErrors, AnalysisDatabase};
 use vm::types::QualifiedContractIdentifier;
 
+#[test]
+fn test_argument_count_violations() {
```

Comment:
Do you want to add some more examples to this?  i.e. for `define-public`, `define-read-only`, and friends?

Reply:
Not sure -- the problem wasn't the `define-` functions -- those arg counts were always checked, since they use `DefineFunctionsParsed::try_parse` which handles the arg count checks (and is covered by tests elsewhere). And other native functions also all never had a problem in read_only_checker, because their arguments are never directly referenced using the array notation (the behavior of those native functions is also covered in other read_only_checker tests).

Rather, it's only the functions that required some special treatment from the read_only_checker that had the issue (`at-block`, `map-get`, and `contract-map-get`.